### PR TITLE
docs: updates application configuration docs

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Adds and updates some Frontend app configuration documents
Organizes configuration object alphabetically

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/docs/application_config.md [hunk 1]
```diff
@@ -17,55 +39,157 @@ To enable this feature, change the `announcements.enabled` boolean value by over
...
+Here is how you would remove the badges list from the homepage
+```js
+browse: {
+  //...
+  showBadgesInHome: false,
+},
+```
+
+## Column Lineage
+*TODO
```
Comment (Golodhros) [inline frontend/docs/application_config.md:125]: @allisonsuarez would love your help for this one, I guess you have it fresh.

File: frontend/docs/application_config.md [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
+      sampleRate: 100
+    }),
+  ],
+}
+```
+
+## Announcements
 
-Annoncements is a feature that allows to disclose new features, changes or any other news to Amundsen's users.
+Annoncements is a feature that allows to disclose new features, changes or any other news to Amundsen's users using a panel in the homepage.
```
Comment (kristenarmes) [inline frontend/docs/application_config.md:31]:
```suggestion
Announcements is a feature that allows to disclose new features, changes or any other news to Amundsen's users using a panel in the homepage.
```

File: frontend/docs/application_config.md [hunk 3]
```diff
@@ -17,55 +39,157 @@ To enable this feature, change the `announcements.enabled` boolean value by over
...
+  enabled: true,
+},
+```
+
+## Badges
 
 Badges are a special type of tag that cannot be edited through the UI.
 
-`BadgeConfig` can be used to customize the text and color of badges. This config defines a mapping of badge name to a `BadgeStyle` and optional `displayName`. Badges that are not defined will default to use the `BadgeStyle.default` style and `displayName` use the badge name with any `_` or `-` characters replaced with a space.
+They [`BadgeStyleConfig` type](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L273) can be used to customize the text and color of badges. This config defines a mapping of badge name to a [`BadgeStyle`](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L264) and optional `displayName`. Badges that are not defined will default to use the `BadgeStyle.default` style and `displayName` use the badge name with any `_` or `-` characters replaced with a space.
```
Comment (kristenarmes) [inline frontend/docs/application_config.md:54]:
```suggestion
The [`BadgeStyleConfig` type](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L273) can be used to customize the text and color of badges. This config defines a mapping of badge name to a [`BadgeStyle`](https://github.com/amundsen-io/amundsen/blob/main/frontend/amundsen_application/static/js/config/config-types.ts#L264) and optional `displayName`. Badges that are not defined will default to use the `BadgeStyle.default` style and `displayName` use the badge name with any `_` or `-` characters replaced with a space.
```

File: frontend/docs/application_config.md [hunk 4]
```diff
@@ -17,55 +39,157 @@ To enable this feature, change the `announcements.enabled` boolean value by over
...
+With it, we configure the max length for table and column descriptions, with defaults being:
+```js
+editableText: {
+  columnDescLength: 250,
+  tableDescLength: 750,
+},
+```
+
+### Examples
+To chage this values, simply update those character lengths:
```
Comment (kristenarmes) [inline frontend/docs/application_config.md:176]:
```suggestion
To change these values, simply update the character lengths:
```

**Review by kristenarmes — APPROVED** [review]
LGTM, just commented on a few small spelling/wordings

**Pull request merged**